src/main/java/org/fcrepo/auth/webac/WebACAuthorizationDelegate.java

```diff
@@ -37,10 +49,45 @@
      */
     private static final Logger LOGGER = LoggerFactory.getLogger(WebACAuthorizationDelegate.class);
 
+    private static final Map<String, URI> actionMap;
+
+    static {
+        Map<String, URI> map = new HashMap<>();
+        map.put("GET", WEBAC_MODE_READ);
+        map.put("POST", WEBAC_MODE_APPEND);
+        map.put("PUT", WEBAC_MODE_WRITE);
+        map.put("DELETE", WEBAC_MODE_WRITE);
+        map.put("PATCH", WEBAC_MODE_WRITE);
+        map.put("OPTIONS", WEBAC_MODE_READ);
+        actionMap = Collections.unmodifiableMap(map);
+    }
+
     @Override
     public boolean rolesHavePermission(final Session userSession, final String absPath,
             final String[] actions, final Set<String> roles) {
-        final boolean permit = false;
+
+        // This is not correct -- we should get it from the container or header, etc
+        final String agent = userSession.getUserID();
+
+        final List<URI> actionURIs = actionsAsURIs(actions);
+
+        final Optional<URI> effectiveACL = getEffectiveAcl(new FedoraResourceImpl(userSession.getNode(absPath)));
```

Comment:
We may need an "internal-session" as the `userSession` will likely not have permissions to read the ACL resources.